### PR TITLE
Fix: #20628: Stop caret left moving off the input string 

### PR DESCRIPTION
**Pull request by Opi-Txm**
Should close [#20628](https://github.com/OpenRCT2/OpenRCT2/issues/20628).

The issue was caused by the code simply not checking whether `selectionOffset == 0` before performing a `selectionOffset--` after a while loop that would possibly run until `selectionOffset` is 0. The code meant to handle this by detecting negative values at the end by comparing the caret's new position with 0 using `std::max<size_t>` and getting the bigger number of it, but since the type of the value is `size_t`, which is unsigned, it blasted the caret away.

The new code simply check if `selectionOffset == 0` before a `selectionOffset--`.
I also removed the negative value detection at the end with since it doesn't make a lot of sense.

I have debugged with this code several times for Japanese, Korean, Vietnamese, English and Simplified Chinese and it looks like it's not going off the edge or does some unexpected behavior.

### CODE REVIEW

**Review by Gymnasiast — COMMENTED** [review]
Code LGTM, needs testing.

**Review by tupaschoal — APPROVED** [review]
Tested it like described in the issue and definitely seems to fix the problem.

**Comment by Opi-Txm** [comment]
Sorry, I put the pr tag in the changelog.txt instead of the issue tag. Just changed it.

File: distribution/changelog.txt [hunk 1]
```diff
@@ -4,6 +4,7 @@
 - Improved: [#20951] Activate OpenRCT2 window after using native file dialog on macOS.
 - Fix: [#20255] Images from the last hovered-over coaster in the object selection are not freed.
 - Fix: [#20616] Confirmation button in the track designer’s quit prompt has the wrong text.
+- Fix: [#20628] Moving caret using Ctrl+left for TTF fonts caused underflow.
```
Comment (Gymnasiast) [inline distribution/changelog.txt:7]:
```suggestion
- Fix: [#20628] Moving caret using Ctrl+left can move too far when using a multibyte grapheme.
```

**Review by Gymnasiast — APPROVED** [review]

**Pull request merged**